### ISSUE TITLE
After tagging, exporting and updating component i can't see changes

### ISSUE THREAD

**Issue by nevolgograd** [teambit/bit#2796]
### Describe the bug

After tagging, exporting and updating component i can't see changes. 
Version in my `package.json` is updated, code in `/node_modules/@bit/scalez.savvy-ui.svg/svg.ts` is updated ,but code in `/node_modules/@bit/scalez.savvy-ui.svg/dist` don't.

I was updating that component not for the first time: tag, export, install as always. Haven’t touched config, also.
Here's my file, it's also updated here - https://bit.dev/scalez/savvy-ui/svg/~code
I was adding instagram icon in last update, and now i can't import it.

Besides that, all my components are in "Generating preview" state. Playground is not loading and I can’t try import icon even there.

### Steps to Reproduce

1. `npm i @bit/scalez.savvy-ui.svg`
2. `import { instagram } from '@bit/scalez.savvy-ui.svg'`
3. `...`
4. `Module '"../../../../../node_modules/@bit/scalez.savvy-ui.svg/dist/svg"' has no exported member 'instagram'.`

### Expected Behavior

Module HAS exported member 'instagram'

### Specifications

- Bit version: 14.8.4
- Node version: 12.16.1
- npm version: 6.14.4
- Platform: Macos
- Bit compiler (include version): compilers/react-typescript/@3.1.52
- Bit tester (include version): -

### Additional context

I've tried to update component once again, tried to delete `node_modules` and `package.json`, but it not helped.  Looks like the issue related to compiler or something.

**Comment by davidfirst** [comment]
@nevolgograd , Regarding this:
> Module HAS exported member 'instagram'

If you look into the files in the dists directory, neither svg.js nor svg.d.ts have the word "instagram". 
Any idea why?

**Comment by nevolgograd** [comment]
@davidfirst it happened on **14.8.0**, i thought bit update will do the thing, but no.

**Comment by nevolgograd** [comment]
@davidfirst you right, it worked! Thanks!
But this is weird, i was successfully updating other components with **14.8.0**, why did it break?

**Comment by davidfirst** [comment]
@nevolgograd , that's what I'm trying to find out. Could you please send me your `.bitmap` file? (you can send to david@bit.dev).

Also, just to make sure, it didn't work with 14.8.0, only with 14.7.6 right?

**Comment by nevolgograd** [comment]
@davidfirst yep, only with 14.7.6

**Comment by davidfirst** [comment]
Thanks @nevolgograd , I got the files and I was able to reproduce the issue. I fixed it in the PR above.

The issue was not with the 'bit build' but with 'bit tag' that triggers the build process.
Because this specific compiler in its latest versions saves `extensions` data into the model, it was confused to be new component and not legacy, as a result, the legacy build process wasn't triggered only the new one (Harmony), which has a different build mechanism.
